### PR TITLE
feat: Utilize official prometheus api for querying SLI values

### LINKED ISSUE #250: Use Prometheus client library to get SLI values
Prometheus project provides an official client library https://github.com/prometheus/client_golang

The library can be used for querying metrics from a prometheus server via the HTTP API (still in alpha stage according the their README).

The idea is to use the client library and see if we can replace our own custom implementation for retrieving SLI values like we do in `GetSLIValue` function https://github.com/keptn-contrib/prometheus-service/blob/6c6e22b0b9e6c52e6fbd079e084f29b704f9cabe/utils/prometheus.go#L196-L248

# Definition of Done

- [ ] SLI values are retrieved correctly
- [ ] Code for crafting metrics requests to Prometheus HTTP API is removed

### PR DESCRIPTION
**Pull request by Raffy23**
## This PR
* Fixes #250

### CODE REVIEW

File: utils/prometheus/prometheus.go [hunk 1]
```diff
@@ -53,7 +40,7 @@ type Handler struct {
 	Service        string
 	DeploymentType string
 	Labels         map[string]string
-	HTTPClient     *http.Client
+	API            apiv1.API
```
Comment (pchila) [inline utils/prometheus/prometheus.go:43]: I know it's an acronym and go requires acronym to be all-uppercase, but here we are also exposing API ehich I don't like much... Can we rename the field to something like

```suggestion
	_API            apiv1.API
```
or
```suggestion
	prometheusAPI            apiv1.API
```

File: utils/prometheus/prometheus.go [hunk 2]
```diff
@@ -196,42 +187,29 @@ func (ph *Handler) GetSLIValue(metric string, start string, end string) (float64
...
+	}
 	if err != nil {
 		return 0, err
 	}
-	defer resp.Body.Close()
 
-	body, _ := ioutil.ReadAll(resp.Body)
-	if resp.StatusCode != 200 {
-		return 0, errors.New("metric could not be received")
+	resultVector := result.(model.Vector)
```
Comment (pchila) [inline utils/prometheus/prometheus.go:201]: Please check the type assertion and handle the (albeit improbable) error :)

```suggestion
	resultVector, ok := result.(model.Vector)
	if !ok {
	    return 0, fmt.Errorf("prometheus response is not a Vector: %v", result)
	}
```

**Review by pchila — APPROVED** [review]
LGTM

**Review by christian-kreuzberger-dtx — APPROVED** [review]
also LGTM from my side :)

**Pull request merged**